src/goto-programs/string_refine_preprocess.cpp

```diff
@@ -0,0 +1,1343 @@
+/*******************************************************************\
+
+Module: Preprocess a goto-programs so that calls to the java String
+        library are recognized by the string solver
+
+Author: Romain Brenguier
+
+Date:   September 2016
+
+\*******************************************************************/
+
+#include "string_refine_preprocess.h"
+
+#include <util/std_expr.h>
+#include <util/symbol_table.h>
+#include <util/pointer_offset_size.h>
+#include <util/prefix.h>
+#include <goto-programs/class_identifier.h>
+#include <solvers/refinement/string_expr.h>
```

Comment:
`string_expr.h` could be moved to util but we need `refined_string_type.h` (this is the type of string_exprt, and maybe these should be renamed to make that obvious). To avoid dependencies on java_bytecode and such, some static of methods of refined_string_type should be moved out of this class to more appropriate places.